### ISSUE TITLE
TravisCI Setup

### ISSUE THREAD

**Issue by JLaveta** [ad440-cloud-project/DevOps#1]
Estimated task time: 8 hours

Investigate/research setup of TravisCI with GitHub/Azure. This task will be considered complete once it can be shown that the Travis CI pipeline is working as intended with new pull requests. This can be demoed using a basic webpage with a text message that will change once a pull request is approved.

**Comment by JLaveta** [comment]
Currently in progress: 3 hours spent on this so far.

Status:

Attempting to follow the guide located here:
https://docs.travis-ci.com/user/deployment/azure-web-apps/

I've setup an AppServicePlan and an AppService. Currently working out the bugs in the TravisCI pipeline. It is currently linked into the GitHub project fine, but I'm not yet able to deploy to Azure.

**Comment by JLaveta** [comment]
Another hour spent working on this without much progress. TravisCI now appears to build the distribution files needed to deploy to Azure, but it is not actually deploying anything to Azure.

4 hours total so far.

**Comment by JLaveta** [comment]
Worked on this another 4 hours, between testing out some fixes to the pipeline and researching.

I have finally connected TravisCI to the webapp resource in Azure, but it is not deploying properly. Currently, it is uploading everything when it really just needs to deploy the dist folder. 

It feels like I'm close here, but I expect this to take another 4 hours or so.

Total time so far: 8 hours.

**Comment by JLaveta** [comment]
Another hour and a half spent trying different solutions with not much progress.

Total time spent: 9.5 hours

**Comment by JLaveta** [comment]
After another hour spent working through this, I've finally made some progress. It looks like TravisCI is still deploying everything to the wwwroot folder in Azure Web Apps. This appears to be causing an issue with how the content is being served.

I manually moved the contents of the dist directory into wwwroot, and the initial Vue website is showing up fine. So, I need to figure out what to change in order to ensure that TravisCI is deploying the correct content to the correct directory.

Cumulative time spent: 10.5 hours

**Comment by JLaveta** [comment]
Though I have not been able to figure out how to do the above, I was able to find a temporary workaround:

In the webapp configuration, I have changed the physical path of the virtual root application path. So, though Travis CI is still pushing the entire repo, at least Azure WebApps knows where to look for the distribution files.

I'll consider this temporarily complete for now, but given the time, I would like to find a better solution. A bug will be opened.

For now, deployed site can be found at: https://ad440-web-app.azurewebsites.net/

Total Time Spent: 11.5
